src/rust/cryptography-x509/src/common.rs

```diff
@@ -281,6 +288,21 @@ pub const PSS_SHA512_MASK_GEN_ALG: MaskGenAlgorithm<'_> = MaskGenAlgorithm {
     params: PSS_SHA512_HASH_ALG,
 };
 
+// From RFC 5480 section 2.1.1:
+// ECParameters ::= CHOICE {
+//     namedCurve         OBJECT IDENTIFIER
+//     -- implicitCurve   NULL
+//     -- specifiedCurve  SpecifiedECDomain }
+//
+// Only the namedCurve form may appear in PKIX. Other forms may be found in
+// other PKIs.
+#[derive(asn1::Asn1Read, asn1::Asn1Write, Hash, Clone, PartialEq, Eq, Debug)]
+pub enum EcParameters<'a> {
+    NamedCurve(asn1::ObjectIdentifier),
+    ImplicitCurve(asn1::Null),
+    SpecifiedCurve(asn1::Tlv<'a>),
```

Comment:
NB: I didn't bother to include the `SpecifiedCurve` definition here, since the original fallback logic didn't do it either. If we wanted to, it's somewhere in X9.62.

Reply:
This should probably be an `asn1::Sequence<'a>` at least, yeah?

Reply:
Yep, that looks like it should be safe per X9.62 6.3:

```asn1
ECParameters ::= SEQUENCE {
version INTEGER { ecpVer1(1) } (ecpVer1),
fieldID FieldID {{FieldTypes}},
curve Curve,
base ECPoint,
order INTEGER,
cofactor INTEGER OPTIONAL,
...
}
```

(Page 20 of <https://safecurves.cr.yp.to/grouper.ieee.org/groups/1363/private/x9-62-09-20-98.pdf>)

Reply:
1a6952dfa..18ad3af8c